src/iHome.Infrastructure.Cache.Extensions/IServiceCollectionExtensions.cs

```diff
@@ -32,4 +32,25 @@ public static IServiceCollection AddCacheFetcher(this IServiceCollection service
 
         return services;
     }
+
+    public static IServiceCollection AddCachedRepository<TInterface, TImplementation>(
+        this IServiceCollection services,
+        Func<IServiceProvider, TImplementation> implementationFactory,
+        Func<ICache, TInterface, TInterface> cacheFactory,
+        ServiceLifetime lifetime = ServiceLifetime.Scoped)
+        where TImplementation : TInterface
```

Comment:
wow generyka

Reply:
w sensie taka ze se podajesz jednak co to ma byc. ciekawe. w javie tego nie ma chyba

Reply:
W javie też da się jakoś extenda lub implementa robić na generyce